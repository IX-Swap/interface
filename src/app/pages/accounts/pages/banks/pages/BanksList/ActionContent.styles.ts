import makeStyles from '@mui/styles/makeStyles'

export const useStyles = makeStyles(theme => ({
  wrapper: {
    padding: theme.spacing(1.5, 2.5),
    background: theme.palette.background.paper,
    borderRadius: 8,
    width: 174,
    boxShadow: `0px 80px 80px ${theme.palette.menu.boxShadow}`,
    border: `1px solid ${theme.palette.select.itemBorder}`,
    opacity: 1
  },
  separator: {
    background: theme.palette.select.itemBorder,
    height: 1,
    margin: theme.spacing(1, 0, 1.5, 0)
  }
}))
