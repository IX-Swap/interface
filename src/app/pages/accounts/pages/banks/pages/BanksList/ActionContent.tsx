import { List, Box } from '@mui/material'
import { DropdownContentProps } from 'app/components/Dropdown/Dropdown'
import React from 'react'
import { useStyles } from 'app/pages/accounts/pages/banks/pages/BanksList/ActionContent.styles'
import { NewAction } from 'app/pages/authorizer/components/NewAction'
export interface ActionContentProps extends DropdownContentProps {
  edit: () => void
  remove: () => void
  view: () => void
}

export const ActionContent = ({
  edit,
  remove,
  view,
  injectedProps
}: ActionContentProps) => {
  const classes = useStyles()
  return (
    <Box className={classes.wrapper}>
      <List data-testid='dropdown' onClick={injectedProps.close}>
        <NewAction label='View details' onClick={view} />
        <div className={classes.separator} />
        <NewAction label='Edit' onClick={edit} />
        <div className={classes.separator} />
        <NewAction label='Delete' onClick={remove} />
      </List>
    </Box>
  )
}
