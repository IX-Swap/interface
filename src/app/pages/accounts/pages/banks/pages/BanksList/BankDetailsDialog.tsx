import { DialogTitle, Grid } from '@mui/material'
import { convertAddressToString } from 'app/pages/authorizer/components/utils'
import { LabelledValue } from 'components/LabelledValue'
import React from 'react'
import { Bank } from 'types/bank'
import { UIDialog } from 'ui/UIDialog/UIDialog'
import { useTheme } from '@mui/material/styles'

export interface BankDetailsDialogProps {
  bank: Bank
  open: boolean
  close: () => void
}

export const BankDetailsDialog = ({
  bank,
  open,
  close
}: BankDetailsDialogProps) => {
  const theme = useTheme()
  return (
    <UIDialog open={open} maxWidth='xs' fullWidth onClose={close}>
      <DialogTitle>Bank Account Information</DialogTitle>
      <Grid container spacing={2}>
        <Grid item xs={12}>
          <Grid container spacing={3} style={{ padding: '20px 40px 40px' }}>
            <Grid item xs={12} md={6}>
              <LabelledValue
                labelColor='gray'
                label='Bank Name'
                value={bank.bankName}
                valueColor={theme.palette.dialog.color}
                labelWeight='thin'
                valueWeight='thin'
              />
            </Grid>
            <Grid item xs={12} md={6}>
              <LabelledValue
                labelColor='gray'
                label='Bank Account Number'
                value={bank.bankAccountNumber}
                valueColor={theme.palette.dialog.color}
                labelWeight='thin'
                valueWeight='thin'
              />
            </Grid>
            <Grid item xs={12} md={6}>
              <LabelledValue
                labelColor='gray'
                label='Swift Code'
                value={bank.swiftCode}
                valueColor={theme.palette.dialog.color}
                labelWeight='thin'
                valueWeight='thin'
              />
            </Grid>
            <Grid item xs={12} md={6}>
              <LabelledValue
                labelColor='gray'
                label='Currency'
                value={bank.currency.symbol}
                valueColor={theme.palette.dialog.color}
                labelWeight='thin'
                valueWeight='thin'
              />
            </Grid>
            <Grid item xs={12} md={6}>
              <LabelledValue
                labelColor='gray'
                label='Bank Address'
                value={convertAddressToString(bank.address)}
                valueColor={theme.palette.dialog.color}
                labelWeight='thin'
                valueWeight='thin'
              />
            </Grid>
            <Grid item xs={12} md={6}>
              <LabelledValue
                labelColor='gray'
                label={`Account Holder's Name`}
                value={bank.accountHolderName}
                valueColor={theme.palette.dialog.color}
                labelWeight='thin'
                valueWeight='thin'
              />
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </UIDialog>
  )
}
