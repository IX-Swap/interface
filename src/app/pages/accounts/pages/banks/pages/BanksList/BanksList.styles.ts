import makeStyles from '@mui/styles/makeStyles'

export const useStyles = makeStyles(theme => ({
  buttonWrapper: {
    background: theme.palette.background.paper,
    padding: theme.spacing(2),
    borderRadius: 8
  },
  pageParent: {
    display: 'table',
    [theme.breakpoints.down('md')]: {}
  }
}))
