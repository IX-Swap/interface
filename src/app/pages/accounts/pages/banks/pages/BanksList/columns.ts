export const columns = [
  {
    key: 'currency.symbol',
    label: 'Currency'
  },
  {
    key: 'bankName',
    label: 'Bank Name'
  },
  {
    key: 'bankAccountNumber',
    label: 'Account Number',
    secret: true
  },
  {
    key: 'accountHolderName',
    label: 'Account Name',
    secret: true
  },
  {
    key: 'swiftCode',
    label: 'Swift Code',
    secret: true
  }
]
