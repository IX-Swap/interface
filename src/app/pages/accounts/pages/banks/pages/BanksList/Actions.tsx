import { Bank } from 'types/bank'
import React, { useState } from 'react'
import { BanksRoute } from 'app/pages/accounts/pages/banks/router/config'
import { Dropdown } from 'app/components/Dropdown/Dropdown'
import { ActionsDropdownTrigger } from 'app/pages/authorizer/components/ActionsDropdownTrigger'
import { ActionContent } from 'app/pages/accounts/pages/banks/pages/BanksList/ActionContent'
import { history } from 'config/history'
import { generatePath } from 'react-router-dom'
import { useRemoveBank } from 'app/pages/accounts/pages/banks/hooks/useRemoveBank'
import { BankDetailsDialog } from 'app/pages/accounts/pages/banks/pages/BanksList/BankDetailsDialog'
import { OTPDialog } from 'app/pages/accounts/pages/banks/pages/WithdrawCash/OTPDialog'
import { Form } from 'components/form/Form'
import { Typography } from '@mui/material'

export interface ActionsProps {
  item: Bank
}

export const Actions = ({ item }: ActionsProps) => {
  const [removeBank, { isLoading }] = useRemoveBank()
  const [showBankDetails, setShowBankDetails] = useState(false)
  const [showOtpDialog, setShowOtpDialog] = useState(false)

  const view = () => {
    setShowBankDetails(true)
  }

  const close = () => {
    setShowBankDetails(false)
  }

  const edit = () =>
    history.push(generatePath(BanksRoute.edit, { bankId: item._id }))

  const remove = async ({ otp }: { otp: string }) => {
    await removeBank({ bankId: item._id, otp })
    closeOtpDialog()
  }

  const openOtpDialog = () => {
    setShowOtpDialog(true)
  }

  const closeOtpDialog = () => {
    setShowOtpDialog(false)
  }

  return (
    <>
      <Dropdown
        arrow
        contentTheme='dark'
        trigger={props => (
          <ActionsDropdownTrigger {...props} isLoading={isLoading} />
        )}
        content={props => (
          <ActionContent
            {...props}
            edit={edit}
            remove={openOtpDialog}
            view={view}
          />
        )}
      />
      <BankDetailsDialog bank={item} open={showBankDetails} close={close} />
      <Form onSubmit={remove}>
        <OTPDialog
          open={showOtpDialog}
          close={closeOtpDialog}
          title='Are You Sure You Want to Remove Bank Account?'
          actionLabel='Confirm'
          content={
            <Typography
              variant='body1'
              align='center'
              style={{ marginBottom: 24 }}
            >
              You will no longer be able to do transactions from this bank
              account
            </Typography>
          }
        />
      </Form>
    </>
  )
}
