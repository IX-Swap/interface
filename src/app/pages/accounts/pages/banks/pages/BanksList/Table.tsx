import React from 'react'
import { columns } from 'app/pages/accounts/pages/banks/pages/BanksList/columns'
import { Bank } from 'types/bank'
import { Actions } from 'app/pages/accounts/pages/banks/pages/BanksList/Actions'
import { useAuth } from 'hooks/auth/useAuth'
import { getIdFromObj } from 'helpers/strings'
import { banksQueryKeys } from 'config/queryKeys'
import { TableView } from 'ui/UIKit/TablesKit/components/TableView/TableView'
export const Table: React.FC = () => {
  const { user } = useAuth()
  const userId = getIdFromObj(user)

  return (
    <TableView<Bank>
      uri={`/accounts/banks/list/${userId}`}
      name={banksQueryKeys.getListByUserId(userId)}
      columns={columns}
      actions={Actions}
    />
  )
}
