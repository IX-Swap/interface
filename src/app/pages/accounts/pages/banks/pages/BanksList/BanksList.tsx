import React from 'react'
import { Grid } from '@mui/material'
import { Header } from 'app/pages/accounts/pages/banks/pages/BanksList/Header'
import { Table } from 'app/pages/accounts/pages/banks/pages/BanksList/Table'
import { VSpacer } from 'components/VSpacer'
import { PageHeader } from 'app/components/PageHeader/PageHeader'
import { RootContainer } from 'ui/RootContainer'

export const BanksList: React.FC = () => {
  return (
    <Grid container direction='column' spacing={2} style={{ display: 'table' }}>
      <Grid item>
        <PageHeader title='Bank Accounts' />
      </Grid>
      <RootContainer>
        <Grid item>
          <Header />
        </Grid>
        <VSpacer size='small' />
        <Grid item style={{ marginLeft: '10px' }}>
          <Table />
        </Grid>
      </RootContainer>
    </Grid>
  )
}
