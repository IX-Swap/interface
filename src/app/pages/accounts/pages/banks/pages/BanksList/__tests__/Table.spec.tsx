import React from 'react'
import { render } from 'test-utils'
import { Table } from 'app/pages/accounts/pages/banks/pages/BanksList/Table'

describe('Table', () => {
  afterEach(async () => {})

  it.skip('renders without error', () => {
    render(<Table />)
  })
})
