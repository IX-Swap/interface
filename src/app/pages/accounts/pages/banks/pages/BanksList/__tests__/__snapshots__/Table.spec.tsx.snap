// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bank Table renders TableView with correct props 1`] = `
<div>
  <form
    style="width: 100%;"
  />
</div>
`;
